src/main/java/duke/command/AddCommand.java

```diff
@@ -0,0 +1,51 @@
+package duke.command;
+
+import duke.main.Ui;
+import duke.utils.Parser;
+import duke.utils.FileStorage;
+import duke.utils.Storage;
+import duke.utils.TaskList;
+
+import java.io.IOException;
+import java.text.ParseException;
+
+/**
+ * Add command implementing command interface
+ */
+public class AddCommand implements Command {
+
+    /**
+     * execute the command by attempting to add task to list and set ui response to
+     * the appropriate one
+     * @param task
+     * @param ui
+     * @param storage
+     * @param taskList
+     */
+    @Override
+    public void execute(String task, Ui ui, Storage storage, TaskList taskList) {
+        String[] token = task.split(" ", 2);
+        if (token.length < 2) {
+            ui.setResponse("Adding task failed, task body cannot be empty");
+            return;
+        }
+        try {
+            if (taskList.addToList(token[1], token[0])) {
+                try {
+                    storage.storeData(Parser.tasksToStorage(taskList.getList()));
+                } catch (IOException e) {
+                    System.out.println("Error in storing data");
```

Comment:
In all of your Exception catching, I see that you only catch the Exception created from Java itself, such as IOException, NumberFormatException, and ParseException. I feel that it is better to create your own class called DukeException, because these are exceptions that are specific to Duke chatbots. So after catching a certain exception thrown by Java, you should know what causes the error, and you can throw another DukeException with the message, for example ("The argument of a done cannot be empty", etc). Then the class that calls this method that gives the exception can catch it and print the error message from DukeException